src/irmin-mem/irmin_mem.ml

```diff
@@ -37,11 +37,15 @@ module Read_only (K : Irmin.Type.S) (V : Irmin.Type.S) = struct
 
   let v _config = Lwt.return map
 
-  let close t =
-    Log.debug (fun f -> f "close");
+  let clear t =
+    Log.debug (fun f -> f "clear");
     t.t <- KMap.empty;
     Lwt.return_unit
 
+  let close t =
+    Log.debug (fun f -> f "close");
+    clear t
```

Comment:
Now that we have a `clear` function, is that the behavior we want (to clear on close?). Maybe trying to be consistent with the other backends makes more sense now. WDYT?

Reply:
My thinking was that in irmin-pack, we are clearing caches when closing, that is close removes all in memory contents. But this is probably not what we want for a memory-only backend.

Reply:
In `irmin-pack`, we are indeed clearing the cache, but that is because they have been flushed to the disk first, no data is removed.